compiler-api/src/main/java/com/squareup/anvil/compiler/api/CodeGenerator.kt

```diff
@@ -28,23 +47,41 @@ internal interface CodeGenerator {
    * last time. Code generates that do not impact other code generators get a last chance to
    * evaluate these results.
    */
-  fun flush(
+  public fun flush(
     codeGenDir: File,
     module: ModuleDescriptor
   ): Collection<GeneratedFile> = emptyList()
 
-  data class GeneratedFile(
+  /**
+   * Represents a generated file that Anvil should eventually write.
+   *
+   * @property file the [File] to generate.
+   * @property content the file contents to write to [file].
+   */
+  public data class GeneratedFile(
```

Comment:
This class needs to be moved into a separate file and not be an inner class. But to blow up this PR not any further, this should be a separate PR. I've added a TODO in #265.